components/local_date_time/local_date_time.jsx

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import PropTypes from 'prop-types';
+import React from 'react';
+
+export default class LocalDateTime extends React.PureComponent {
+    static propTypes = {
+
+        /*
+         * The time to display
+         */
+        eventTime: PropTypes.number,
+
+        /*
+         * Set to display using 24 hour format
+         */
+        useMilitaryTime: PropTypes.bool,
+
+        /*
+         * Current timezone of the user
+         */
+        timezone: PropTypes.string,
+    };
+
+    render() {
+        const {
+            eventTime,
+            timezone,
+            useMilitaryTime,
+        } = this.props;
+
+        const date = eventTime ? new Date(eventTime) : new Date();
+
+        const options = {
+            hour: '2-digit',
+            minute: '2-digit',
+            hour12: !useMilitaryTime,
+        };
+
+        if (timezone) {
+            options.timeZone = timezone;
+        }
+
+        return (
+            <time
+                className='post__time'
+                dateTime={date.toISOString()}
+                title={date}
+            >
+                {date.toLocaleString('en', options)}
```

Comment:
Could you replace this with a FormattedTime component from react-intl like I was going to do [here](https://github.com/mattermost/mattermost-webapp/commit/633c3e43fa9bb2fd350e54882dc3ece906a71c48)? It takes those same options as props, but it'll be slightly more performant since it caches some stuff internally